cedar-policy-generators/src/hierarchy.rs

```diff
@@ -457,6 +457,11 @@ impl<'a, 'u> HierarchyGenerator<'a, 'u> {
                         let num_entities_per_type = num_entities / entity_types.len();
                         let mut uids = HashSet::new();
                         while uids.len() < num_entities_per_type {
+                            // If we run out of bytes in `u`, then `uid` will be the same on every
+                            // subsequent iteration, so the size of `uids` won't increase.
+                            if self.u.is_empty() {
+                                return Err(Error::NotEnoughData);
+                            }
```

Comment:
Might make sense to add this error handling code to `generate_uid_with_type` instead.

Reply:
I tend to agree, but, `generate_uid_with_type` does have a way to proceed (rather than erroring) if it doesn't have data in `u`, and maybe it is worth preserving that

Reply:
I'd guess that proceeding with a default value instead of erroring out should generally be the preferred behavior since it throws away fewer inputs